### PR TITLE
FIX: [droid] disable libcec

### PR DESCRIPTION
**Pull request by koying**
Library was never shipped, and would never work anyway

/cc @MartijnKaijser @Razzeee @fritsch

### CODE REVIEW

**Comment by MartijnKaijser** [comment]
@opdenkamp ping just to be safe

**Comment by opdenkamp** [comment]
why?

**Comment by MrMC** [comment]
because it requires kernel support that is not present in any android kernel :)

**Comment by koying** [comment]
Was the "why" re the PR?
All droids handle CEC in firmware, and even if some wouldn't, there is no standard interface

**Comment by opdenkamp** [comment]
so just `cdc_acm` not enabled? meaning that flipping the flag when compiling magically makes it work? :)
@koying yes. problem with the vendor's kernel implementations is that they simply don't work except if you're lucky enough to have bought a TV from a sane vendor

**Comment by MrMC** [comment]
sure, one could flip that bit when compiling but why even carry the load of an addon that is never used in the first place ?

**Comment by opdenkamp** [comment]
standard kernel interface is on it's way btw, but I haven't had time to check the state of that recently. though knowing how most android vendors work, they probably won't see it working before 2024

**Comment by koying** [comment]
Never said their CEC implem was working ;)
Re cdc_acm, I don't know what it is, but, generally speaking, Android enabled selinux on recent versions, so even going to sysfs directly would probably be a no-go

**Comment by opdenkamp** [comment]
s/never used/never used by most people/ ;-)
`cdc_acm` is the kernel module needed to talk to the adapter. serial over usb emulation.
I don't use this myself really, just saying that there's nothing in libCEC preventing it from working on android afaict. Assuming that permissions are set up correctly and kernel stuff is enabled. What's the advantage of removing support from our side except for saving a couple of KB in the binary?

**Comment by opdenkamp** [comment]
and (just to colour the bikeshed red ;-)): you should really make your commit message more descriptive than what you've put there now. it's not a fix and the reason for removing it, the one that you've put in the PR's description, should really be in the commit message.

**Comment by MrMC** [comment]
I prefer green :)

**Comment by MrMC** [comment]
I saw some thread, somewhere. There was a libcec.so placed in actual firmware and it collided with the shipped libcec.so. Think it was a crash&burn(tm) too. Same libs, maybe, who knows.

**Comment by koying** [comment]
> it's not a fix and the reason for removing it, the one that you've put in the PR's description, should really be in the commit message

It's a fix because there is an issue, now, apparently (see my edit re the forum post above)
And the PR comment IS in the commit.

Really hard to have even obvious stuff merged those days... 
<bikeshed mode OFF>

**Comment by opdenkamp** [comment]
@koying my bad, github chose to hide that part. but if you think that me asking why support for something that I wrote gets pulled from android is too much for you, then don't PR. just asking for an explanation, that's all.

if it actually fixes a crash, sure, go ahead. if it's just saving a few KB and nothing is actually fixed by removing this, then I don't really see the point.

**Comment by koying** [comment]
> collided with the shipped libcec.so

That's the point. We never shipped our own in the first place on droid (by omission, I guess). So whatever lib called "libcec.so" on the system would *always* be loaded, whatever it actually is/do.

@opdenkamp Alternative is to actually ship our libcec.so ;)
Now, I personally don't see the point of shipping something that, afaik, will never work, or at best will interfere with whatever shitty implem is in the firmware, but up to you, guys...

**Comment by opdenkamp** [comment]
right, that makes more sense. fine with me to pull the compiler flag if it was never in and tries to load a random version of the lib. perhaps add that that was happening to the commit message though, to make it clear what was causing the problem, more than just "would never work anyway". nobody but you will know what the reason was when checking the log in a year or so.

**Comment by koying** [comment]
@opdenkamp Actually, it would be great if you could tell me how you think it could/should work on Android at all ;)

**Comment by opdenkamp** [comment]
it would either need one of our CEC adapter, or support for the in-kernel stuff, like this https://github.com/Pulse-Eight/libcec/pull/198 (once finished/working, though knowing aml, it will work until the next chip is out)
for our adapter it needs the `cdc_acm` module enabled in the kernel.
other than that, no differences to the other platforms afaik

**Comment by koying** [comment]
Oh, is P8 actually working on stock Android? That's totally different, then...

**Comment by opdenkamp** [comment]
well it's just linux with a gimped libc and (a lot of) userspace fluff, isn't it? ;-)

**Comment by koying** [comment]
Let me rephrase: Did you ever succeed in having a P8 working on a stock Android, i.e. without a homebrewed kernel? ;)

**Comment by opdenkamp** [comment]
haven't checked, but if `cdc_acm` is not included by any of the android stbs, as @MrMC pointed out, then it's not going to work very well

**Comment by MartijnKaijser** [comment]
Conclusion is? Patch it out in Jarvis regardless and in master as well with comment to revise later when it works with updated libcec?

**Comment by opdenkamp** [comment]
42. whatever the android maintainer decides. oooh... wait... ;)

**Comment by MrMC** [comment]
hahahaha

**Comment by MartijnKaijser** [comment]
I expect a proper response not some childish behaviour

**Comment by MrMC** [comment]
ok, two previous android maintainers suggest this would be good to go in, and would also be safe for jarvis backport. It does not touch the other platforms and only targets Android. Seems logical that someone should hit the green button and do the backport to jarvis.

**Comment by stefansaraev** [comment]
jenkins build this please. just in case.

**Comment by fritsch** [comment]
Also not an android maintainer, but i am with @MrMC and @koying

**Comment by opdenkamp** [comment]
... which is the conclusion that you could have drawn if you'd actually read what was being discussed here. so much negative energy around here nowadays, are you still doing this for fun? :)

**Comment by FernetMenta** [comment]
> ok, two previous android maintainers suggest this would be good to go in

would be great if previous would be finite too. guys, you decided not to contribute anymore. why don't you just stop your negative contribution too?

**Comment by MrMC** [comment]
meh, just who voted you lord god of kodi development ? I will do as I please thank you very much and that includes commenting on PRs should I feel they are worth the effort.

**Comment by opdenkamp** [comment]
sorry, but what's negative about this contribution?

**Comment by fritsch** [comment]
I would prefer that the previous would be the current and I would prefer it further if we could come along somehow. Everyone needs to cope with reality and it's sadly clear after some months without a maintainer that there won't be someone coming out of nowhere that cares just a tiny little bit about android technically or would contribute code, but the ex maintainers do and contribute, which I really like.

I still think that upstream kodi has the most potential for future releases as it can do better in certain disciplines that in combination produce a higher value in usability, features and so on. On the other hand we have seen multiple android specific backports during the last weeks, which will improve v17's Android experience a lot.

**Comment by retamiro** [comment]
Guys, excuse me for resurrect this issue , but the problem reported in http://forum.kodi.tv/showthread.php?tid=265630&page=2 is simply because they are not sending the libcec the apk , I always saw this error in the log earlier, about the possible conflict with any existing libcec would a matter of versioning and decision which libcec use , type the latest ?
Now I use an android stb ( Tegra 2 Android 4.0.3 ) and never used the libcec , never tried a matter of priority.
I think in part to contribute to Kodi , but chat me discouraged and did not have the experience of others , but I'm getting to backport the kodi for Tegra 2 and Android 4.0.3 .
Do not let die the project because the kodi is simply amazing.
